File: app/src/main/java/com/discovr/discord/injection/main/CardFragmentComponent.java
package com.discovr.discord.injection.main;

import com.discovr.discord.injection.util.scope.FragmentScope;
import com.discovr.discord.ui.main.card.CardFragment;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@FragmentScope
@Subcomponent(modules = { CardFragmentModule.class })
public interface CardFragmentComponent extends AndroidInjector<CardFragment> {
    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<CardFragment> {}
}

File: app/src/main/java/com/discovr/discord/injection/main/CardFragmentComponent.kt
package com.discovr.discord.injection.main

import com.discovr.discord.injection.util.FragmentScope
import com.discovr.discord.ui.main.card.CardFragment

import dagger.Subcomponent
import dagger.android.AndroidInjector

@FragmentScope
@Subcomponent(modules = arrayOf(CardFragmentModuleBind::class, CardFragmentModuleProvide::class))
interface CardFragmentComponent : AndroidInjector<CardFragment> {
    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<CardFragment>()
}